File: rx-android-utils/src/main/java/com/utils/rxandroid/exceptions/AppException.java
package com.utils.rxandroid.exceptions;

import android.support.annotation.StringRes;

import lombok.Getter;

/**
 * Created by Nick Unuchek on 21.12.2017.
 */

public class AppException extends RuntimeException {
    @StringRes @Getter private int messageResId;
    public AppException() {
        super();
    }
    public AppException(@StringRes int messageResId) {
        super();
        this.messageResId = messageResId;
    }

    public AppException(String message) {
        super(message);
    }

    public AppException(String message, Throwable cause) {
        super(message, cause);
    }

    public AppException(Throwable cause) {
        super(cause);
    }
}

File: rx-android-utils/src/main/java/com/utils/rxandroid/exceptions/AppException.kt
package com.utils.rxandroid.exceptions

import android.support.annotation.StringRes

/**
 * Created by Nick Unuchek on 21.12.2017.
 */

class AppException : RuntimeException {
    @StringRes
    var messageResId: Int = -1

    constructor() : super()

    constructor(@StringRes messageResId: Int) : super() {
        this.messageResId = messageResId
    }

    constructor(message: String) : super(message)

    constructor(message: String, cause: Throwable) : super(message, cause)

    constructor(cause: Throwable) : super(cause)
}
